src/core/caniuse.js

```diff
@@ -0,0 +1,212 @@
+    conf.caniuse = { feature: conf.caniuse, ...DEFAULTS };
+    return;
+  }
+  if (Array.isArray(conf.caniuse.browsers)) {
+    conf.caniuse.browsers = conf.caniuse.browsers.map(b => b.toLowerCase());
+  } else if (conf.caniuse.browsers !== "ALL") {
+    conf.caniuse.browsers = DEFAULTS.browsers;
+  }
+  if (conf.caniuse.maxAge === undefined) conf.caniuse.maxAge = DEFAULTS.maxAge;
+  if (!conf.caniuse.versions) conf.caniuse.versions = DEFAULTS.versions;
+}
+
+/**
+ * main canIUse function
+ * @param  {String} key              which api to look for
+ * @param  {Node} refNode            adds table after refNode
+ * @param {IDBCache} cache           IDB cache
+ * @param {Object} conf              normalized respecConfig.caniuse
+ */
+async function canIUse(key, refNode, cache, conf) {
+  const title = hyperHTML`
+    <dt class="caniuse-title" id="${`caniuse-${key}`}">
+      Can I Use this API? (${key})
+    </dt>`;
+
+  const contentPromise = new Promise(async (resolve) => {
+    // use data from cache data if valid and render
+    try {
+      const cachedStats = await cache.get(key);
+      if (cachedStats
+        && new Date() - new Date(cachedStats.cacheTime) < conf.maxAge) {
+        resolve(getTableHtml(key, cachedStats.stats, conf));
+      }
+    } catch (err) {
+      console.error(err);
+    }
+
+    // otherwise fetch new data, cache and render
+    const url = conf.apiURL
+      ? conf.apiURL.replace("{FEATURE}", key)
+      : `https://raw.githubusercontent.com/Fyrd/caniuse/master/features-json/${key}.json`;
+
+    try {
+      const json = await getJson(url);
+      cache.set(key, { stats: json.stats, cacheTime: new Date() })
+        .catch(err => console.error("Failed to cache caniuse data.", err));
+      resolve(getTableHtml(key, json.stats, conf));
+    } catch (err) {
+      const permalink = `http://caniuse.com/#feat=${key}`;
+      resolve(hyperHTML`
+        Error [core/caniuse]: ${err.message}.
+        <br>Please check directly on <a href="${permalink}">${permalink}</a>.`);
+    }
+  });
+
+  const content = hyperHTML`
+    <dd class="caniuse-stats">${{
+      any: contentPromise,
+      placeholder: "fetching data from caniuse.com...",
+    }}</dd>`;
+
+  refNode.parentNode.insertBefore(content, refNode.nextSibling);
+  refNode.parentNode.insertBefore(title, refNode.nextSibling);
+}
+
+// TODO: replace with fetch in core/github ?
+async function getJson(url) {
+  const response = await window.fetch(url);
+  if (!response.ok) {
+    if (response.status === 404) {
+      console.error(`The resource ${url} could not be found (HTTP 404)`);
+      throw new Error("Could not fetch GitHub resource (HTTP 404)");
+    }
+    const errorMsg = `GitHub Response not OK. Probably exceeded request limit. (HTTP ${response.status})`;
+    console.error(`${errorMsg}. Resource = ${url}`);
+    throw new Error(errorMsg);
+  }
+  return await response.json();
+}
+
+const BROWSERS = { // browser name dictionary
```

Comment:
This should be a `new Map()`:

```JS
const BROWSERS = new Map([
  ["chrome", "Chrome"],
  ["firefox", "Firefox"],
  ["ie", "IE"],
  ["edge", "Edge"],
  ["android", "Android"],
  ["safari", "Safari"],
  ["opera", "Opera"],
  ["bb", "Blackberry"],
  ["and_uc", "UC (Android)"],
  ["and_ff", "Firefox (Android)"],
]);
```

Because:

```JS
conf.browsers: ["hasOwnProperty"]; // or anything in the prototype chain
```

Would return true when you do the "in" check.

Reply:
Nit: move to the top of the file